File: app2/src/main/java/com/ssz/framejava/base/CustomActivity.java
package com.ssz.framejava.base;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.Nullable;

import butterknife.ButterKnife;

@SuppressLint("Registered")
public abstract class CustomActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        beforeOnCreate();
        super.onCreate(savedInstanceState);
        int layoutId = getLayoutId();
        if (layoutId != 0){
            setContentView(layoutId);
            ButterKnife.bind(this);
        }
        initView();
        initData();
    }

    protected void beforeOnCreate(){

    }
    abstract int getLayoutId();
    abstract void initView();
    abstract void initData();

}

File: app/src/main/java/com/ssz/base/CustomActivity.kt
package com.ssz.base

import android.os.Bundle

/**
 * @author : zsp
 * time : 2019 10 2019/10/24 16:29
 */
abstract class CustomActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        beforeOnCreate()
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        initView()
        initData()
    }

    protected open fun beforeOnCreate(){

    }
    abstract fun getLayoutId():Int
    abstract fun initView()
    abstract fun initData()
}